frontend-react/src/pages/submissions/SubmissionDetails.tsx

```diff
@@ -104,6 +104,20 @@ function SubmissionDetailsContent() {
     );
     const submissionDate = generateSubmissionDate(actionDetails.submittedAt);
```

Comment:
generateSubmissionDate can return `SubmissionDate | null`, so you need protections below when you access it `${submissionDate.dateString} ${submissionDate.timeString}` -> `${submissionDate?.dateString || ''} ${submissionDate?.timeString || ''}`

Or you can change generateSubmissionDate to never return null?

Reply:
The check is there, but in the form of the inline if/else:
```typescript
const titleString = submissionDate ? 
        `${submissionDate.dateString} ${submissionDate.timeString}`
        : 
        "Date and Time parsing error";
```
This is because the `SubmissionDate` type cannot have null or undefined values if returned:
```typescript
type SubmissionDate = {
    dateString: string;
    timeString: string;
};
```

Reply:
Can you add a couple of tests for SubmissionDate?
```typescript
test("SubmissionDate returns null for invalid blank strings", () => {
    const submissionDate = generateSubmissionDate("");
    expect(submissionDate).toBe(null);
});

test("SubmissionDate returns null for invalid number strings", () => {
    const submissionDate = generateSubmissionDate("-1");
    expect(submissionDate).toBe(null);
});
```

I'm getting:
```text
  ● SubmissionDate returns null for invalid number strings

    expect(received).toBe(expected) // Object.is equality

    Expected: null
    Received: {"dateString": "1 Jan 2001", "timeString": "12:0 AM"}

      41 | test("SubmissionDate returns null for invalid number strings", () => {
      42 |     const submissionDate = generateSubmissionDate("-1");
    > 43 |     expect(submissionDate).toBe(null);
         |                            ^
      44 | });
      45 |
```

"Writing correct datetime functions is hard. Making them work for localization is even harder." -me

Reply:
Oh, interesting! 🤔 Is there a reason a Date considers -1 to be Jan 1 2020 at "12:0". I guess a better question is does it make more sense to protect against short inputs, negative inputs, or both?

Reply:
Yeah, doing your own date formatting has lots of corner cases and localization issues. As soon as you get a "we need it in military time" or "It needs to show the timezone", you'll reconsider rolling your own. 😜 
And as much as I dislike the bloat of the moment library, there's a reason it's so big (and popular).

But I'll mark this issue resolved. You can open a new bug if you think it's required.

Reply:
I would caution against Moment.js: https://momentjs.com/docs/#/-project-status/

> We now generally consider Moment to be a legacy project in maintenance mode. It is not dead, but it is indeed done.

Reply:
Agreed. I've been replacing it where possible. I think this project still uses it in a few places. (maybe create a bug for it?)

The correct way to format is probably to use the newer `Intl.DateTimeFormat()` feature and accept that you may not get the _exact_ format you wanted?